src/frontend/Updater.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+    Copyright 2016-2020 Arisotura, WaluigiWare64
+    This file is part of melonDS.
+    melonDS is free software: you can redistribute it and/or modify it under
+    the terms of the GNU General Public License as published by the Free
+    Software Foundation, either version 3 of the License, or (at your option)
+    any later version.
+    melonDS is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+    FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License along
+    with melonDS. If not, see http://www.gnu.org/licenses/.
+*/
+
+#include <iostream>
+#include <string>
+#include <cstring> 
+#include <curl/curl.h>
+#include <nlohmann/json.hpp>
+#include <cstdio>
+
+namespace Updater
+{
+using json = nlohmann::json;
+static size_t WriteCallback(void *contents, size_t size, size_t nmemb, void *userp)
+{
+    ((std::string*)userp)->append((char*)contents, size * nmemb);
+    return size * nmemb;
+}
+size_t write_data(void *ptr, size_t size, size_t nmemb, FILE *stream) {
+    size_t written = fwrite(ptr, size, nmemb, stream);
+    return written;
+}
+std::string getCURL(const char* webURL, const char* authFlag="", bool fwrite=false) 
+{
+    CURL *curl;
+    CURLcode res;
+    FILE *fp;
+  
+    std::string readBuffer;
+
+    curl = curl_easy_init();
+    if(curl) 
+    {
+        curl_easy_setopt(curl, CURLOPT_URL, webURL);
+        curl_easy_setopt(curl, CURLOPT_USERAGENT, "curl/7.77.1");

```

Comment:
Is `7.77.1` a hardcoded version string?

Reply:
yep it's a version string.
the GitHub API wouldn't work if I didn't have a useragent.

Reply:
Then use the actual version string in libcurl headers or something, hardcoding is not good practice.

Reply:
done.